tests/integration/cocoon/test_cocoon_format_instruments.py

```diff
@@ -0,0 +1,104 @@
+import unittest
+import uuid
+from pathlib import Path
+
+import lusid
+import pandas as pd
+
+from lusidtools import cocoon as cocoon
+from lusidtools import logger
+
+
+class CocoonTestsFormatInstruments(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+        cls.logger = logger.LusidLogger("debug")
+
+        cls.mapping_required = {"name": "instrument_name"}
+        cls.mapping_optional = {
+            "look_through_portfolio_id.scope": "lookthrough_scope",
+            "look_through_portfolio_id.code": "lookthrough_code",
+        }
+        cls.identifier_mapping = {
+            "Instrument/default/Figi": "figi",
+            "Instrument/default/ClientInternal": "client_internal",
+        }
+
+    def test_all_fields(self):
+        scope = f"TestScope-{uuid.uuid4()}"
+
+        portfolios = pd.DataFrame(
+            {"code": "Child", "name": "Child Portfolio", "currency": "GBP", "date": "01/01/2020"},
+            {"code": "Parent", "name": "Parent Portfolio", "currency": "GBP", "date": "01/01/2020"}
+        )
+
+        cocoon.cocoon.load_from_data_frame(
```

Comment:
Can we check here that the portfolios do get upserted here? If this request fails then the validation below will fail, so it would be better to raise an error here in this case